sklearn/ensemble/stacking.py

```diff
@@ -0,0 +1,269 @@
+"""Stacking API"""
+
+# Author: Caio Oliveira <caioaao@gmail.com>
+# License: BSD 3 clause
+
+from ..base import (BaseEstimator, TransformerMixin, MetaEstimatorMixin)
+from ..model_selection import cross_val_predict
+from ..pipeline import (_name_estimators, FeatureUnion)
+from ..preprocessing import FunctionTransformer
+
+
+class StackingTransformer(BaseEstimator, MetaEstimatorMixin, TransformerMixin):
+    """Transformer to turn estimators into meta-estimators for model stacking
+
+    In stacked generalization, meta estimators are combined in layers to
+    improve the final result. To prevent data leaks between layers, a procedure
+    similar to cross validation is adopted, where the model is trained in one
+    part of the set and predicts the other part. In `StackingTransformer`, it
+    happens during `fit_transform`, as the result of this procedure is what
+    should be used by the next layers.
+
+    Parameters
+    ----------
+    base_estimator : the estimator to be blended.
+
+    cv : cv to be used, optional (default=3)
+        Will be passed to `cross_val_predict` during `fit_transform`.
+
+    method : string, optional (default='auto')
+        Invokes the passed method name of the passed estimator. If the method
+        is `auto`, will try to invoke, for each estimator, `predict_proba`,
+        `decision_function` or `predict` in that order.
+
+    n_jobs : int, optional (default=1)
+        Number of jobs to be passed to `cross_val_predict` during
+        `fit_transform`.
+
```

Comment:
Add a Notes section highlighting the difference between fit_transform(X) and transform(X)